internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -105,6 +106,10 @@ func create(engine *EngineOperations, rpcOps *client.RPC, pid int) error {
 		c.suidFlag = 0
 	}
 
+	if !c.userNS {
+		c.userNS, _ = namespaces.ParentUserNamespace(os.Getpid())
```

Comment:
Please add a comment here. I'm guessing something like "even if the use of a user namespace was not requested by the user, we are already running in a user namespace, so set things up as if the user requested that mode of operation".

Or whatever is right. I'm guessing.